src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,106 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
+@SuppressWarnings("unchecked")
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double DEFAULT_MULTIPLIER = 1.5;
+    private int capacity;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        capacity = DEFAULT_CAPACITY;
+        elementData = (T[]) new Object[capacity];
+    }
+
+    private void resizeIfNeeded() {
```

Comment:
please move private methods after public because inner logic that comes before public methods that allow to get understanding of a class looks confusing

Reply:
not fixed
All private methods should be below public methods